### PR TITLE
[huajiao] Add new extractor

### PR DESCRIPTION
**Pull request by pyx**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This is the extractor for http://www.huajiao.com/ (Huajiao 花椒直播), a live video streaming site.

### CODE REVIEW

File: youtube_dl/extractor/huajiao.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        feed = self._parse_json(feed_json, video_id)
+
+        description = self._html_search_meta(
+            'description', webpage, 'description', fatal=False)
+
+        return {
+            'id': video_id,
+            'title': feed['feed']['formated_title'],
+            'description': description,
+            'duration': parse_duration(feed['feed']['duration']),
```
Comment (yan12125) [inline youtube_dl/extractor/huajiao.py]: For optional fields like ```duration```, ```feed['feed']['duration']``` should be ```feed.get('feed', {}).get('duration')```. All fields other than ``id```, ```title``` and ```formats``` are optional.

Comment (pyx) [inline youtube_dl/extractor/huajiao.py]: Changed as required. Please review. Thank you.

**Comment by yan12125** [comment]
Thanks for the extractor! Would you like to be listed in [AUTHORS](https://github.com/rg3/youtube-dl/blob/master/AUTHORS)? If so under what name?

**Comment by pyx** [comment]
Yes, it would be my honor. Philip Xu.
Thank you.

**Comment by yan12125** [comment]
Done!

**Pull request merged**